src/main/kotlin/controller/MineSweeperController.kt

```diff
@@ -1,14 +1,28 @@
 package controller
 
-import domain.Board
+import domain.MineSweeperGame
 import ui.InputView
 import ui.ResultView
 
 object MineSweeperController {
     fun run() {
         val gameData = InputView.askInfo()
-        val board = Board(gameData)
+        val mineSweeperGame = MineSweeperGame(gameData)
 
-        ResultView(gameData).printMineGameState(board.squares)
+        val resultView = ResultView(gameData)
+        resultView.printStart()
+
+        do {
+            resultView.printBoard(mineSweeperGame.board)
+
+            val openPosition = InputView.askOpenPosition()
+            mineSweeperGame.open(openPosition)
+        } while (!mineSweeperGame.hasDone)
+
+        if (mineSweeperGame.hasLose) {
+            resultView.printLose()
+        } else {
+            resultView.printWin()
+        }
```

Comment:
이 부분은 hasLose boolean 정보를 ResultView에 넘겨 아래와 같이 처리해주는건 어떨까요?

```suggestion
resultView.printResult(mineSweeperGame.hasLose)
```